src/client.cpp

```diff
@@ -281,91 +377,36 @@ void CClient::OnNewConnection()
-{
-    // translate from server channel IDs to client channel IDs
-    // ALSO here is where we allocate and free client channels as required
-
-    const int iNumConnectedClients = vecChanInfo.Size();
-    int       i, iSrvIdx;
-
-    // on a new connection, a server sends an empty channel list before sending
-    // the real channel list (see #1010). To avoid this discarding "our" channel
-    // that we have just created, we skip this processing and just pass the empty
-    // list to the emitted signal.
 
-    if ( iNumConnectedClients != 0 )
+    // Reset all gains for unused channel IDs:
+    for ( int iId = 0; iId < MAX_NUM_CHANNELS; iId++ )
     {
-        // this relies on the received client list being in order of server channel ID
-
-        for ( i = 0, iSrvIdx = 0; i < iNumConnectedClients && iSrvIdx < MAX_NUM_CHANNELS; )
-        {
-            // server channel ID of this entry
-            const int iServerChannelID = vecChanInfo[i].iChanID;
-
-            // find matching client channel ID, creating new if necessary,
-            // update channel number to be client-side
-            vecChanInfo[i].iChanID = FindClientChannel ( iServerChannelID, true );
-
-            // discard any lower server channels that are no longer in our local list
-            while ( iSrvIdx < iServerChannelID )
-            {
-                const int iId = FindClientChannel ( iSrvIdx, false );
-
-                if ( iId != INVALID_INDEX )
-                {
-                    // iSrvIdx contains a server channel number that has now gone
-                    FreeClientChannel ( iSrvIdx );
-                }
-                iSrvIdx++;
-            }
-
-            i++;       // next list entry
-            iSrvIdx++; // next local server channel
-        }
-
-        // have now run out of active channels, discard any remaining from our local list
-        // note that iActiveChannels will reduce as remaining channels are freed
-
-        while ( iActiveChannels > iNumConnectedClients && iSrvIdx < MAX_NUM_CHANNELS )
+        if ( !bChanIdInUse[iId] )
         {
-            const int iId = FindClientChannel ( iSrvIdx, false );
-
-            if ( iId != INVALID_INDEX )
-            {
-                // iSrvIdx contains a server channel number that has now gone
-                FreeClientChannel ( iSrvIdx );
-            }
-            iSrvIdx++;
+            // reset oldGain and newGain as this channel id is currently unused and will
+            // start with a server-side gain at 1 (100%) again.
+            oldGain[iId] = newGain[iId] = 1;
         }
-
-        Q_ASSERT ( iActiveChannels == iNumConnectedClients );
     }
 
-    // pass the received list onwards, now containing client channel IDs
-
-    emit ConClientListMesReceived ( vecChanInfo );
+    // update mixer board with the additional client infos
+    audioMixerBoard.ApplyNewConClientList( vecChanInfo );
+    // set session status
+    setSessionStatus("CONNECTED");
```

Comment:
I don’t think we need to necessarily always have a text saying „Connected“. The state should be conveyed differently. 
if the connection stands faders are visible, else not. Potential we‘d want to say „Disconnected“ in this case.